source/slang/slang-type-layout.h

```diff
@@ -658,6 +658,9 @@ class EntryPointLayout : public ScopeLayout
     // The shader profile that was used to compile the entry point
     Profile profile;
 
+    // The name of the entry point. Always available even if entryPoint is nullptr (for example when it came from a library)
+    Name* name = nullptr;
```

Comment:
There's an interesting question here of whether this is best stored as a `Name` or just a plain old string, but I suppose there is no harm in using a `Name`.